# Assignment 1
MD Sakif Al Mohaimen \
ID - 201861408 \
Date - 23/1/2023

In [1]:
import pandas as pd
df = pd.read_csv('starwars.csv')
df

id seen_any_films  fan seen_episode1 seen_episode2  \
0     3.292880e+09            Yes  Yes           Yes           Yes   
1     3.292880e+09             No  NaN           NaN           NaN   
2     3.292765e+09            Yes   No           Yes           Yes   
3     3.292763e+09            Yes  Yes           Yes           Yes   
4     3.292731e+09            Yes  Yes           Yes           Yes   
...            ...            ...  ...           ...           ...   
1181  3.288389e+09            Yes  Yes           Yes           Yes   
1182  3.288379e+09            Yes  Yes           Yes           Yes   
1183  3.288375e+09             No  NaN           NaN           NaN   
1184  3.288373e+09            Yes  Yes           Yes           Yes   
1185  3.288373e+09            Yes   No           Yes           Yes   

     seen_episode3 seen_episode4 seen_episode5 seen_episode6  rank_episode1  \
0              Yes           Yes           Yes           Yes            3.0   
1              NaN           NaN           NaN           NaN            NaN   
2              Yes            No            No            No            1.0   
3              Yes           Yes           Yes           Yes            5.0   
4              Yes           Yes           Yes           Yes            5.0   
...            ...           ...           ...           ...            ...   
1181           Yes           Yes           Yes           Yes            5.0   
1182           Yes           Yes           Yes           Yes            4.0   
1183           NaN           NaN           NaN           NaN            NaN   
1184           Yes           Yes           Yes           Yes            4.0   
1185            No            No           Yes           Yes            6.0   

      ...           view_yoda                    who_shot_first  \
0     ...      Very favorably  I don't understand this question   
1     ...                 NaN                               NaN   
2     ...    Unfamiliar (N/A)  I don't understand this question   
3     ...      Very favorably  I don't understand this question   
4     ...  Somewhat favorably                            Greedo   
...   ...                 ...                               ...   
1181  ...      Very favorably                               Han   
1182  ...      Very favorably  I don't understand this question   
1183  ...                 NaN                               NaN   
1184  ...      Very favorably                               Han   
1185  ...    Very unfavorably  I don't understand this question   

      familiar_with_expanded_universe  fan_expanded_universe  fan_star_trek  \
0                                 Yes                     No             No   
1                                 NaN                    NaN            Yes   
2                                  No                    NaN             No   
3                                  No                    NaN            Yes   
4                                 Yes                     No             No   
...                               ...                    ...            ...   
1181                               No                    NaN            Yes   
1182                               No                    NaN            Yes   
1183                              NaN                    NaN             No   
1184                               No                    NaN            Yes   
1185                               No                    NaN             No   

      gender    age     household_income                         education  \
0       Male  18-29                  NaN                High school degree   
1       Male  18-29         $0 - $24,999                   Bachelor degree   
2       Male  18-29         $0 - $24,999                High school degree   
3       Male  18-29  $100,000 - $149,999  Some college or Associate degree   
4       Male  18-29  $100,000 - $149,999  Some college or Associate degree   
... 

# Answer 1
Number of unique values for age.

In [2]:
age = df['age'].nunique()
age

4

# Answer 2
Unique responses for 'Who shot first?'.

In [3]:
df['who_shot_first'].unique()

array(["I don't understand this question", nan, 'Greedo', 'Han'],
      dtype=object)

# Answer 3

In [4]:
df.columns

Index(['id', 'seen_any_films', 'fan', 'seen_episode1', 'seen_episode2',
       'seen_episode3', 'seen_episode4', 'seen_episode5', 'seen_episode6',
       'rank_episode1', 'rank_episode2', 'rank_episode3', 'rank_episode4',
       'rank_episode5', 'rank_episode6', 'view_han_solo',
       'view_luke_skywalker', 'view_princess_leia_organa',
       'view_anakin_skywalker', 'view_obi_wan_kenobi',
       'view_emperor_palpatine', 'view_darth_vader', 'view_lando_calrissian',
       'view_boba_fett', 'view_c3p0', 'view_r2d2', 'view_jar_jar_binks',
       'view_padme_amidala', 'view_yoda', 'who_shot_first',
       'familiar_with_expanded_universe', 'fan_expanded_universe',
       'fan_star_trek', 'gender', 'age', 'household_income', 'education',
       'location'],
      dtype='object')

In [5]:
df1 = df[(df['view_darth_vader'] == 'Very favorably') | (df['view_darth_vader'] == 'Somewhat favorably')]
df1['view_darth_vader']

0           Very favorably
3           Very favorably
4       Somewhat favorably
5           Very favorably
6           Very favorably
               ...        
1175        Very favorably
1180        Very favorably
1181    Somewhat favorably
1184        Very favorably
1185        Very favorably
Name: view_darth_vader, Length: 481, dtype: object

In [6]:
df1

id seen_any_films  fan seen_episode1 seen_episode2  \
0     3.292880e+09            Yes  Yes           Yes           Yes   
3     3.292763e+09            Yes  Yes           Yes           Yes   
4     3.292731e+09            Yes  Yes           Yes           Yes   
5     3.292719e+09            Yes  Yes           Yes           Yes   
6     3.292685e+09            Yes  Yes           Yes           Yes   
...            ...            ...  ...           ...           ...   
1175  3.288401e+09            Yes  Yes           Yes           Yes   
1180  3.288390e+09            Yes  Yes           Yes           Yes   
1181  3.288389e+09            Yes  Yes           Yes           Yes   
1184  3.288373e+09            Yes  Yes           Yes           Yes   
1185  3.288373e+09            Yes   No           Yes           Yes   

     seen_episode3 seen_episode4 seen_episode5 seen_episode6  rank_episode1  \
0              Yes           Yes           Yes           Yes            3.0   
3              Yes           Yes           Yes           Yes            5.0   
4              Yes           Yes           Yes           Yes            5.0   
5              Yes           Yes           Yes           Yes            1.0   
6              Yes           Yes           Yes           Yes            6.0   
...            ...           ...           ...           ...            ...   
1175           Yes           Yes           Yes           Yes            1.0   
1180           Yes           Yes           Yes           Yes            3.0   
1181           Yes           Yes           Yes           Yes            5.0   
1184           Yes           Yes           Yes           Yes            4.0   
1185            No            No           Yes           Yes            6.0   

      ...           view_yoda                    who_shot_first  \
0     ...      Very favorably  I don't understand this question   
3     ...      Very favorably  I don't understand this question   
4     ...  Somewhat favorably                            Greedo   
5     ...      Very favorably                               Han   
6     ...      Very favorably                               Han   
...   ...                 ...                               ...   
1175  ...  Somewhat favorably                               Han   
1180  ...      Very favorably                               Han   
1181  ...      Very favorably                               Han   
1184  ...      Very favorably                               Han   
1185  ...    Very unfavorably  I don't understand this question   

      familiar_with_expanded_universe  fan_expanded_universe  fan_star_trek  \
0                                 Yes                     No             No   
3                                  No                    NaN            Yes   
4                                 Yes                     No             No   
5                                 Yes                     No            Yes   
6                                 Yes                     No             No   
...                               ...                    ...            ...   
1175                              Yes                     No            Yes   
1180                               No                    NaN             No   
1181                               No                    NaN            Yes   
1184                               No                    NaN            Yes   
1185                               No                    NaN             No   

      gender    age     household_income                         education  \
0       Male  18-29                  NaN                High school degree   
3       Male  18-29  $100,000 - $149,999  Some college or Associate degree   
4       Male  18-29  $100,000 - $149,999  Some college or Associate degree   
5       Male  18-29    $25,000 - $49,999                   Bachelor degree   
6       Male  18-29                  NaN                High school degree   
... 

In [7]:
len(df1)

481

As we can see in 'view_darth_vader', 'Very favorably' or 'Somewhat favorably' have appeared 481 times.

# Answer 4
Number of people that have seen all six episodes of star wars.

In [8]:
df2 = df[(df['seen_episode1'] == 'Yes') & (df['seen_episode2'] == 'Yes') & (df['seen_episode3'] == 'Yes') & (df['seen_episode4'] == 'Yes') & (df['seen_episode5'] == 'Yes') & (df['seen_episode6'] == 'Yes')]
df2

id seen_any_films  fan seen_episode1 seen_episode2  \
0     3.292880e+09            Yes  Yes           Yes           Yes   
3     3.292763e+09            Yes  Yes           Yes           Yes   
4     3.292731e+09            Yes  Yes           Yes           Yes   
5     3.292719e+09            Yes  Yes           Yes           Yes   
6     3.292685e+09            Yes  Yes           Yes           Yes   
...            ...            ...  ...           ...           ...   
1175  3.288401e+09            Yes  Yes           Yes           Yes   
1180  3.288390e+09            Yes  Yes           Yes           Yes   
1181  3.288389e+09            Yes  Yes           Yes           Yes   
1182  3.288379e+09            Yes  Yes           Yes           Yes   
1184  3.288373e+09            Yes  Yes           Yes           Yes   

     seen_episode3 seen_episode4 seen_episode5 seen_episode6  rank_episode1  \
0              Yes           Yes           Yes           Yes            3.0   
3              Yes           Yes           Yes           Yes            5.0   
4              Yes           Yes           Yes           Yes            5.0   
5              Yes           Yes           Yes           Yes            1.0   
6              Yes           Yes           Yes           Yes            6.0   
...            ...           ...           ...           ...            ...   
1175           Yes           Yes           Yes           Yes            1.0   
1180           Yes           Yes           Yes           Yes            3.0   
1181           Yes           Yes           Yes           Yes            5.0   
1182           Yes           Yes           Yes           Yes            4.0   
1184           Yes           Yes           Yes           Yes            4.0   

      ...           view_yoda                    who_shot_first  \
0     ...      Very favorably  I don't understand this question   
3     ...      Very favorably  I don't understand this question   
4     ...  Somewhat favorably                            Greedo   
5     ...      Very favorably                               Han   
6     ...      Very favorably                               Han   
...   ...                 ...                               ...   
1175  ...  Somewhat favorably                               Han   
1180  ...      Very favorably                               Han   
1181  ...      Very favorably                               Han   
1182  ...      Very favorably  I don't understand this question   
1184  ...      Very favorably                               Han   

      familiar_with_expanded_universe  fan_expanded_universe  fan_star_trek  \
0                                 Yes                     No             No   
3                                  No                    NaN            Yes   
4                                 Yes                     No             No   
5                                 Yes                     No            Yes   
6                                 Yes                     No             No   
...                               ...                    ...            ...   
1175                              Yes                     No            Yes   
1180                               No                    NaN             No   
1181                               No                    NaN            Yes   
1182                               No                    NaN            Yes   
1184                               No                    NaN            Yes   

      gender    age     household_income                         education  \
0       Male  18-29                  NaN                High school degree   
3       Male  18-29  $100,000 - $149,999  Some college or Associate degree   
4       Male  18-29  $100,000 - $149,999  Some college or Associate degree   
5       Male  18-29    $25,000 - $49,999                   Bachelor degree   
6       Male  18-29                  NaN                High school degree   
... 

In [9]:
len(df2)

471

# Answer 5
Percentage of males and females that are fan of star wars.

In [10]:
df['gender'].value_counts()

Female    549
Male      497
Name: gender, dtype: int64

In [11]:
male = df[(df['gender'] == 'Male')]
male_fan = male[(male['fan'] == 'Yes')]
percentage_of_male_fan = (len(male_fan) / len(df)) * 100
percentage_of_male_fan

25.54806070826307

In [12]:
female = df[(df['gender'] == 'Female')]
female_fan = female[(female['fan'] == 'Yes')]
percentage_of_female_fan = (len(female_fan) / len(df)) * 100
percentage_of_female_fan

20.06745362563238